Fw/Tlm/TlmPacket.cpp

```diff
@@ -10,91 +10,148 @@
 
 namespace Fw {
 
-    TlmPacket::TlmPacket() : m_id(0) {
+    TlmPacket::TlmPacket() : m_numEntries(0) {
         this->m_type = FW_PACKET_TELEM;
+        this->m_tlmBuffer.resetSer();
     }
 
     TlmPacket::~TlmPacket() {
     }
 
-    SerializeStatus TlmPacket::serialize(SerializeBufferBase& buffer) const {
-        SerializeStatus stat;
-#if !FW_AMPCS_COMPATIBLE
-        stat = serializeBase(buffer);
-        if (stat != FW_SERIALIZE_OK) {
+    SerializeStatus TlmPacket::resetPktSer() {
+        this->m_tlmBuffer.resetSer();
+        // reset packet count
+        this->m_numEntries = 0;
+        // serialize descriptor
+        // The function serializeBase inherited from ComPacket converts this->m_type
+        // to type FwPacketDescriptorType and serializes the result into this->m_tlmBuffer.
```

Comment:
Should we check or assert that this->m_type is FW_PACKET_TELEM? Otherwise this->m_tlmBuffer will be in a bad state.

Reply:
Added an assert. It should never change since we set it in the constructor and don't modify it (we never deserialize from a buffer for downlink), but it can't hurt.

Reply:
Sounds goo!